File: MyApplication/app/src/main/java/com/homework/myapplication/adapter/ResultAdapter.java
package com.homework.myapplication.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.homework.myapplication.R;
import com.homework.myapplication.bean.Score;

import java.util.List;


public class ResultAdapter extends RecyclerView.Adapter<ResultAdapter.ResultHolder> {
    private Context context;
    private List<Score> scoreList;
    private OnItemClickListener onItemClickListener;

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    public interface OnItemClickListener{
        void onItemClick(View v,int postion);
    }

    public ResultAdapter(Context context, List<Score> scoreList) {
        this.context = context;
        this.scoreList = scoreList;
    }

    @NonNull
    @Override
    public ResultHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_result,parent,false);
        return new ResultHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ResultHolder holder, int position) {
        Score score = scoreList.get(position);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(context);
        holder.recyclerView.setLayoutManager(linearLayoutManager);
        holder.recyclerView.setHasFixedSize(true);
        QuestionAdapter questionAdapter = new QuestionAdapter(context,score.getExamList());
        holder.recyclerView.setAdapter(questionAdapter);
        if (score.isShow()){
            holder.recyclerView.setVisibility(View.VISIBLE);
        }else {
            holder.recyclerView.setVisibility(View.GONE);
        }
        holder.tvName.setText("Name: "+score.getUserName());
        holder.tvResult.setText(" Grade: "+score.getScore());
        holder.tvDate.setText(" Test Date: "+score.getExamDate());
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemClickListener.onItemClick(v,position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return scoreList.size();
    }

    public class ResultHolder extends RecyclerView.ViewHolder {
        private TextView tvResult;
        private TextView tvName;
        private TextView tvDate;
        private RecyclerView recyclerView;

        public ResultHolder(@NonNull View itemView) {
            super(itemView);
            tvResult = itemView.findViewById(R.id.tv_result);
            tvName = itemView.findViewById(R.id.tv_name);
            tvDate = itemView.findViewById(R.id.tv_date);
            recyclerView = itemView.findViewById(R.id.recycler_view);
        }

    }
}

File: MyApplication/app/src/main/java/com/homework/myapplication/adapter/ResultAdapter.kt
package com.homework.myapplication.adapter

import android.app.Application
import com.homework.myapplication.model.DaoManager
import com.homework.myapplication.interfaces.CallBack
import com.homework.myapplication.bean.Score
import android.app.Activity
import android.content.Intent
import com.homework.myapplication.controller.ResultActivity
import com.homework.myapplication.app.MyApplication
import com.homework.myapplication.bean.Exam
import com.homework.myapplication.bean.Answer
import com.homework.myapplication.util.SharePerferenceUtils
import com.homework.myapplication.util.DataUtil
import com.homework.myapplication.bean.Question
import com.homework.myapplication.util.ListUtil
import android.database.sqlite.SQLiteOpenHelper
import com.homework.myapplication.util.DataBaseHelper
import android.database.sqlite.SQLiteDatabase
import android.content.SharedPreferences
import android.content.ContentValues
import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.homework.myapplication.R
import android.text.TextUtils
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.widget.TextView
import com.homework.myapplication.adapter.ResultAdapter.ResultHolder
import androidx.recyclerview.widget.LinearLayoutManager
import com.homework.myapplication.adapter.QuestionAdapter
import androidx.fragment.app.FragmentPagerAdapter
import android.os.Bundle
import kotlin.jvm.JvmOverloads
import androidx.appcompat.app.AppCompatActivity
import com.homework.myapplication.controller.NoScrollViewPager
import com.homework.myapplication.util.GuidGeneratorUtil
import com.homework.myapplication.controller.ExamFragment
import com.homework.myapplication.adapter.MyFragmentPagerAdapter
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import android.widget.Toast
import com.homework.myapplication.util.PecentUtil
import android.os.Build
import com.homework.myapplication.adapter.OptionAdapter
import android.widget.EditText
import com.homework.myapplication.controller.ExamActivity
import com.homework.myapplication.adapter.ResultAdapter
import androidx.viewpager.widget.ViewPager
import android.view.MotionEvent
import android.view.View

class ResultAdapter(private val context: Context, private val scoreList: List<Score?>?) :
    RecyclerView.Adapter<ResultHolder>() {
    private var onItemClickListener: OnItemClickListener? = null
    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?) {
        this.onItemClickListener = onItemClickListener
    }

    interface OnItemClickListener {
        fun onItemClick(v: View?, postion: Int)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ResultHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_result, parent, false)
        return ResultHolder(view)
    }

    override fun onBindViewHolder(holder: ResultHolder, position: Int) {
        val score = scoreList!![position]
        val linearLayoutManager = LinearLayoutManager(context)
        holder.recyclerView.layoutManager = linearLayoutManager
        holder.recyclerView.setHasFixedSize(true)
        val questionAdapter = QuestionAdapter(context, score.getExamList())
        holder.recyclerView.adapter = questionAdapter
        if (score!!.isShow) {
            holder.recyclerView.visibility = View.VISIBLE
        } else {
            holder.recyclerView.visibility = View.GONE
        }
        holder.tvName.text = "Name: " + score.userName
        holder.tvResult.text = " Grade: " + score.score
        holder.tvDate.text = " Test Date: " + score.examDate
        holder.itemView.setOnClickListener { v -> onItemClickListener!!.onItemClick(v, position) }
    }

    override fun getItemCount(): Int {
        return scoreList!!.size
    }

    inner class ResultHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvResult: TextView
        val tvName: TextView
        val tvDate: TextView
        val recyclerView: RecyclerView

        init {
            tvResult = itemView.findViewById(R.id.tv_result)
            tvName = itemView.findViewById(R.id.tv_name)
            tvDate = itemView.findViewById(R.id.tv_date)
            recyclerView = itemView.findViewById(R.id.recycler_view)
        }
    }
}